bundles/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/IhcBinding.java

```diff
@@ -396,28 +399,79 @@ private void updateResource(String itemName, Type type, boolean updateOnlyOutBin
 			
 			try {
 
-				int resourceId = provider.getResourceId(itemName);
-				WSResourceValue value = ihc
-						.getResourceValueInformation(resourceId);
+				int resourceId = provider.getResourceId(itemName, (Command)type);
+				logger.trace(
+						"found resourceId {} (item='{}', state='{}', class='{}')",
+						new Object[] { new Integer(resourceId).toString(), itemName, type.toString(),
+								type.getClass().toString() });
+				if(resourceId > 0) {
 				
-				ArrayList<IhcEnumValue> enumValues = null;
-
-				if (value instanceof WSEnumValue) {
-
-					enumValues = ihc.getEnumValues(((WSEnumValue) value)
-							.getDefinitionTypeID());
-				}
-
-				value = IhcDataConverter.convertCommandToResourceValue(type, value,
-						enumValues);
+					WSResourceValue value = ihc
+							.getResourceValueInformation(resourceId);
+					
+					ArrayList<IhcEnumValue> enumValues = null;
+	
+					if (value instanceof WSEnumValue) {
+	
+						enumValues = ihc.getEnumValues(((WSEnumValue) value)
+								.getDefinitionTypeID());
+					}
+	
+					Integer val = provider.getValue(itemName, (Command)type);
+					boolean trigger = false;
+					if(val != null) {
+						if(val == 0) {
+							type = OnOffType.OFF;
+						} else if(val == 1) {
+							type = OnOffType.ON;
+						} else {
+							trigger = true;
+						}
+					}
+					
+					if (!trigger) {
+						value = IhcDataConverter.convertCommandToResourceValue(type, value,
+								enumValues);
 
-				boolean result = updateResource(value);
+						boolean result = updateResource(value);
+						
+						if (result == true) {
+							logger.debug("Item updated '{}' succesfully sent",
+									itemName);
+						} else {
+							logger.error("Item '{}' update failed", itemName);
+						}
+					} else {
+						value = IhcDataConverter.convertCommandToResourceValue(OnOffType.ON, value,
+								enumValues);
+		
+						boolean result = updateResource(value);
+						
+						if (result == true) {
+							logger.debug("Item '{}' trigger started",
+									itemName);
+							
+							Thread.sleep(val);
```

Comment:
This sleep duration probably should be limited? Long sleep will block another resource id updates.